### PR TITLE
feat: changed state passing from React context to Event Manager

### PR DESCRIPTION
**Pull request by danial031193**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,10 +51,8 @@ import Layout from './components/layout';
 import Dialog from './modals/default';
 
 const App = () => {
```
Comment (MatthewPattell) [inline README.md]:
```
const App = () => (
  <>
    <Layout />
    <ModalRoot Modal={(props) => <Dialog {...props} />} />
  </>
)
```

File: src/root.tsx [hunk 2]
```diff
@@ -1,32 +1,137 @@
...
 import type { FC } from 'react';
-import React from 'react';
-import { ModalConsumer } from './context';
-import type { IModalRoot } from './types';
+import React, { useEffect, useCallback, useState } from 'react';
+import CHANNEL from './channel';
+import type { IEventsPayload, IModalItem, IModalProps, IModalRoot, THideModal } from './types';
 
 /**
  * ModalRoot Consumer
```
Comment (MatthewPattell) [inline src/root.tsx]:
```suggestion
 * ModalRoot 
```

File: src/types.ts [hunk 3]
```diff
@@ -1,58 +1,51 @@
...
-  hide: ReturnType<typeof useModal<TProps>>[1];
+export interface IModalProps<TCP extends object> {
+  hookRef?: IModalHookRef<TCP>;
 }
 
-export interface IModalProps<TProps extends object = Record<string, any>> extends IModalToggle {
-  children?: ReactNode;
-  hookRef?: IModalHookRef<TProps>;
-}
+export type OmitStoreProps<T> = {
```
Comment (MatthewPattell) [inline src/types.ts]: https://github.com/Lomray-Software/react-mobx-manager/blob/prod/src/with-stores.tsx#L16

File: src/channel.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+enum CHANNEL {
+  OPEN = 'OPEN',
+  HIDE = 'HIDE',
```
Comment (MatthewPattell) [inline src/channel.ts]:
```suggestion
  OPEN = 'react-modals:open',
  HIDE = 'react-modals:hide',
```

**Review by MatthewPattell — CHANGES_REQUESTED** [review]

**Comment by MatthewPattell** [comment]
Support multimodal?

**Comment by danial031193** [comment]
> Support multimodal?

Yes

**Review by MatthewPattell — APPROVED** [review]

**Pull request merged**